numpy/add_newdocs.py

```diff
@@ -1928,12 +1928,19 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     dot(a, b, out=None)
 
-    Dot product of two arrays.
+    Dot product of two arrays. Specifically,
```

Comment:
One more case needed - if either of `a` or `b` are 0-D/scalars, then elementwise multiplication is used (`np.multiply`)

Reply:
Is this behavior that we want to legitimitize by documenting?

Obviously, we won't change it at this point, but I would still encourage users to use `*` over dot in every circumstance.

Reply:
@shoyer: Isn't that true of this whole list? Isn't `*` or `@` always preferable over dot?

Reply:
Hmm. Can we explicitly recommend using matmul or times instead of dot in the docstring here then?

Reply:
@shoyer Please see my updates. Thanks!